### DISCUSSION TITLE
Constant shadowing in function parameters

### DISCUSSION THREAD

**Discussion by SleepModezZ** [vlang/v#15179] [General]
Hello, I have a question concerning constants in V. Is it by design that constants are allowed to be shadowed by local variables (without warnings) in situations where the local variable is a function parameter? For example:

```
const number = 1

fn test(number int) {   // Shadowing but no warning?
    println(number) 
}

fn main() {
    test(2)
}
```
(The program prints 2.)

I accidentally coded something similar to this after I read about constants in V and converted a little C-program of mine to V.

 (The program worked just fine, but I was a bit confused with my code. In C the constant would have been "#define NUMBER 1" - and I never use ALL_CAPS_NAMES for local variables - so there would not have been any confusion with the identifiers. But I can't use ALL_CAPS_NAMES for constants in V. )

Thank you!

**Comment by JalonSolov** [comment]
That should fail as it is.  There shouldn't be _any_ shadowing allowed.  Otherwise it starts getting just as confusing as in any other language... "Oh, did you mean the constant, the variable, or the function named 'number'?"
